File: examples/python/core/parallel/result-ordering.py
from aws_durable_execution_sdk_python import (
    BatchResult,
    DurableContext,
    durable_execution,
)

@durable_execution
def handler(event: dict, context: DurableContext) -> list[str]:
    """Demonstrate result ordering."""
    
    functions = [
        lambda ctx: ctx.step(lambda _: "First"),
        lambda ctx: ctx.step(lambda _: "Second"),
        lambda ctx: ctx.step(lambda _: "Third"),
    ]
    
    result = context.parallel(functions)
    
    # Results are in the same order as functions
    results = result.get_results()
    assert results[0] == "First"
    assert results[1] == "Second"
    assert results[2] == "Third"
    
    return results

File: examples/java/core/parallel/result-ordering.java
var future1 = ctx.runInChildContextAsync("first", String.class,
    child -> child.step("work", String.class, stepCtx -> "First"));

var future2 = ctx.runInChildContextAsync("second", String.class,
    child -> child.step("work", String.class, stepCtx -> "Second"));

var future3 = ctx.runInChildContextAsync("third", String.class,
    child -> child.step("work", String.class, stepCtx -> "Third"));

DurableFuture.allOf(future1, future2, future3);

// Results are in the same order as futures
var results = List.of(future1.get(), future2.get(), future3.get());
assert results.get(0).equals("First");
assert results.get(1).equals("Second");
assert results.get(2).equals("Third");

return Map.of("results", results);
